Update roadmap to clarify large data and workflows examples; add multi select groups and drilldown (#384)
holoviz/holoviz: doc/Roadmap.rst

    Plotly, Matplotlib)**: Datashader provides its own "shading" support, turning
    an array into colors for display, but if shading is done by an external
    plotting program, the plotting program can support user-friendly features
    like hover information, legends, and colorbars. There is already extensive
    support for rendering Datashader output in Bokeh, along with some support in
    Matplotlib and Plotly. The core team is gradually further improving Bokeh's
    support, with nearly all of Datashader's shading functionality now or soon
    available interactively in Bokeh, and similar support can be added to other
    libraries by their users or contributors.

3.  **Support for managing annotations**: Users of a plot or dashboard often
    notice important features of the data on display, but don't have a convenient
    way to capture or communicate that insight. The core team is developing
    annotations as a first-class component the HoloViz tools, easily added onto
    plots and dashboards so that users can denote regions of interest in some
    space (e.g. a time range on a timeseries plot), storing, restoring, and

>     displaying that information in editable form.

Reviewer: ```suggestion
    displaying that information in editable form. See https://github.com/holoviz/holonote for the work as it develops.
```

> Other targets needed in HoloViz or in packages designed to work well with

Reviewer: ```suggestion
Other goals for HoloViz or packages designed to work well with
```

    many other ways to set up live Python-backed plots and Panel dashboards or
    apps, including creating a Flask server or embedding into Django, along with
    deploying via Heroku or Google Cloud and AWS or via the Anaconda Enterprise
    platform or on `MyBinder <https://mybinder.org>`__. Documenting and testing
    these possibilities takes time and effort, and any contributions or examples
    that will help people get started and decide between the alternatives would
    be very helpful.

2.  **Bokeh WebGL support**: Bokeh provides good support for working
    interactively with small datasets using an HTML Canvas (client-side
    interactivity), and when combined with Datashader it can handle enormous
    datasets by pre-rendering them to images in Python (server-side). However, in
    between these two extremes there is a range of data sizes that could be
    addressed well by the client-side WebGL technology. Bokeh has long included
    some WebGL support, but improvements in 2021-2022 have made the WebGL support
    much more solid and usable, and updating HoloViews to use that could bring

>     big performance improvements for midsize plots.

Reviewer: @Hoxbro , can you confirm -- can we now delete this WebGL entry? I'm pretty sure this has been completed and WebGL is now the default for HoloViews.

Reviewer 2: WebGL is now the default in HoloViews.

Author: removed

> 3.  **New and improved workflow examples in biomedical and Earth science,

Reviewer: ```suggestion
3.  **New and improved workflow examples in biomedical and Earth sciences,
```

(to me the plural reads better, since biomedical science is a separate field from Earth science, but it's just a preference).

    especially utilizing machine learning**: Demonstrations for handling and
    visualization of various large, multidimensional datasets from data-intensive
    research domains, including on HPC systems.

4.  **Better deployment and exploration of large-scale visualizations**: For
    instance, with improvements to automatic generation of tiles for exploration

>     of large datasets using standard web servers.

Reviewer: Maybe link to the last version of those examples on Datashader that got deleted, to point people to what to pick up?

Author: this? https://datashader.org/tiling.html

Reviewer: The better link would be https://github.com/holoviz/datashader/blob/main/examples/tiling.ipynb , but for some reason I thought it had been deleted recently.

Reviewer 3: Yes we currently discussed this notebook (and discovered it personally) as we were listing the S3 buckets we use and it's pointing to one. Jim you said:

> Datashader's tiling support is in exactly the state you are observing: a proof of concept, untested, unlinked in the docs, and yet still potentially useful for someone making tiles and discoverable by someone really dedicated. If rioxarray has better support, great! We should then make an example of using rioxarray with Datashader to render the tiles. If there is no open issue about dealing with tiles, there definitely should be; Brendan indeed left it in a half-finished state. I wouldn't worry about the S3 bucket, as it would appear to be for dumping the output, not holding precious input files.  So I'm inclined to do no action here besides having a better issue open if there isn't one, as we don't currently have any funding for tile-related work.

Author: link to nb added

    focused primarily on statistical visualizations, not covering areas like
    multidimensional (image-like) plotting where HoloViews and Bokeh are also
    used, but they offer a common interchange format that could be helpful for
    interoperating with other tools. Writing an import and export facility that
    covers the bulk of the shared functionality should not be a major undertaking
    and could open up interesting new applications. For now, Altair and Vega-lite
    visualizations can be specified and then used directly with Panel, but not
    with HoloViews.

10. **hvPlot/HoloViews serialization**: HoloViews uses a declarative design
    that can be represented in a purely textual form, without any Python code. An
    initial implementation allows any Param-based objects (including HoloViews
    objects) to be represented in JSON or YAML, but it needs some polishing
    before it can be put into wide use for saving and restoring configurations
    and layouts.

11. **Multiple selection groups**: HoloViews currently supports a single
    selection group. Multiple selection groups would allow for more flexible use

>     of, and comparison between, selections.

Reviewer: I think this has to be explained better, as I don't think "selection group" is a common term in our docs.

Author: text revised

> 12. **Intuitive drilldown support**: In a drilldown action, the user
>     interacts in some way with one main plot (usually some low-dimensional

Reviewer: ```suggestion
12. **Easy to configure drilldown support**: In a drilldown action, the user
    interacts in some way with one main plot (usually some low-dimensional
```
